### PR TITLE
Add metrics for istio-agent startup time

### PR DESCRIPTION
**Pull request by howardjohn**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by howardjohn** [comment]
@istio/wg-policies-and-telemetry-maintainers you are not owners but probably should have some eyes on this

File: pilot/cmd/pilot-agent/metrics/metrics.go
Comment (douglas-reid) [inline pilot/cmd/pilot-agent/metrics/metrics.go:32]: curious: do you want to track this by proxy version or anything similar?

File: pilot/cmd/pilot-agent/metrics/metrics.go
Comment (douglas-reid) [inline pilot/cmd/pilot-agent/metrics/metrics.go:30]: instead of just `_time`, can you make this `startup_duration_seconds` or whatever the appropriate unit is.

File: pilot/cmd/pilot-agent/metrics/metrics.go
Comment (douglas-reid) [inline pilot/cmd/pilot-agent/metrics/metrics.go:29]: is this the kind of thing we care about aggregating across a group of proxies?  or is this just to look at a few individual results in perf monitoring ?

File: pilot/cmd/pilot-agent/metrics/metrics.go
Comment (howardjohn) [reply pilot/cmd/pilot-agent/metrics/metrics.go:32]: How is this normally done? It seems like *any* metric we might want to.. but is it something that should be done by prom or higher level tool? Not sure if it has a way to "join" the pod name or other label and fetch the label

File: pilot/cmd/pilot-agent/metrics/metrics.go
Comment (howardjohn) [reply pilot/cmd/pilot-agent/metrics/metrics.go:30]: yep

File: pilot/cmd/pilot-agent/metrics/metrics.go
Comment (howardjohn) [reply pilot/cmd/pilot-agent/metrics/metrics.go:29]: I think individual ones can be useful to see outliers?

Comment (ramaraochavali) [reply pilot/cmd/pilot-agent/metrics/metrics.go:29]: Should we use "initialization_time_ms" stat of Envoy https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/statistics.html?highlight=server%20uptime#server?

Not against adding a new stat but proxy already measures it - so just thinking if it is worth reusing it?

Comment (howardjohn) [reply pilot/cmd/pilot-agent/metrics/metrics.go:29]: I am not sure they are measuring the same thing. let me compare them

**Comment by howardjohn** [comment]
They are fairly similar. Technically they are measuring different things since the agent readiness is not the same as envoys readiness, so it may be helpful to have both.

**Comment by ramaraochavali** [comment]
sure. No issues in having additional metric

**Review by ramaraochavali — COMMENTED** [review]
LGTM.

**Comment by howardjohn** [comment]
@ramaraochavali can you give github approval or anything you want me to change/get reviewed by someone else?

**Comment by ramaraochavali** [comment]
Sure. I was thinking if @douglas-reid has any other questions. I will approve and if he has any questions we can do a follow-up

**Review by ramaraochavali — APPROVED** [review]

File: pilot/cmd/pilot-agent/metrics/metrics.go
Comment (douglas-reid) [reply pilot/cmd/pilot-agent/metrics/metrics.go:32]: the only reason i asked is that this metric is proxy-specific -- and the agent is possibly able to know the version at startup time (and could record it directly possibly).

File: releasenotes/notes/agent-startup.yaml
Comment (douglas-reid) [inline releasenotes/notes/agent-startup.yaml:7]: need to update name here to match the code.

**Review by adiprerepa — APPROVED** [review]

**Review by douglas-reid — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Pull request merged**
